proxy/start_container_interceptor.go

```diff
@@ -2,16 +2,41 @@ package proxy
 
 import (
 	"net/http"
+
+	"github.com/fsouza/go-dockerclient"
 )
 
 type startContainerInterceptor struct{ proxy *Proxy }
 
+type startContainerRequestBody struct {
+	HostConfig *docker.HostConfig `json:"HostConfig,omitempty" yaml:"HostConfig,omitempty"`
+}
+
 func (i *startContainerInterceptor) InterceptRequest(r *http.Request) error {
 	container, err := inspectContainerInPath(i.proxy.client, r.URL.Path)
-	if err == nil && containerShouldAttach(container) {
+	if err != nil || !containerShouldAttach(container) {
+		return err
+	}
+
+	if r.Header.Get("Content-Type") != "application/json" {
 		i.proxy.createWait(r, container.ID)
+		return nil
 	}
-	return err
+
+	hostConfig := &docker.HostConfig{}
+	if err := unmarshalRequestBody(r, &hostConfig); err != nil {
+		return err
+	}
+
+	i.proxy.addWeaveWaitVolume(hostConfig)
+	if dnsDomain, withDNS := i.proxy.getDNSDomain(); withDNS {
+		if err := i.proxy.setWeaveDNS(hostConfig, container.Config.Hostname, dnsDomain); err != nil {
```

Comment:
I don't think so. `dnsDomain` is the (possibly) untrimmed version `weave.local.`, while the `container.Config.Domainname` gets the trailing dot trimmed off. We only set it if the user has not specified `--name`, `--hostname`, or `--dns-search`.